### ISSUE TITLE
Add support for mixed distance and duration objectives

### ISSUE THREAD

**Issue by N-Wouda** [PyVRP/PyVRP#543]
#### Describe the solution you'd like

We currently only support an objective that minimises distance and some penalty terms for the hard constraints. Minimising duration, or more specifically, a weighted combination of distance and duration, is also very common and desirable. We should probably make the weights vehicle type specific. This should be pretty easy to add in now that we have centralised the relevant cost computations.

#### Describe alternatives you have considered

None, this is unsupported.

**Comment by N-Wouda** [comment]
@leonlan since you know more about HFVRP, would this get us HFVRP in full generality? Or does that also require changes to the actual dist/duration matrices, like in #475?

**Comment by N-Wouda** [comment]
Yeah, looking at this comment [here](https://github.com/PyVRP/PyVRP/issues/188#issuecomment-1565369152) I think it does.

**Comment by leonlan** [comment]
Yea by just having vehicle-dependent weights we can solve HFVRP.

**Comment by N-Wouda** [comment]
OK I'll have a look at this soon. It should not be hard to do.

**Comment by N-Wouda** [comment]
@leonlan the sandbox contains some HFVRP instances. Now that #546 is essentially done, shall we check check those and upgrade them to the instances repository?

**Comment by leonlan** [comment]
@N-Wouda I'll upgrade them to PyVRP/Instances tomorrow and make the necessary changes to `read.py` as well!

**Comment by N-Wouda** [comment]
Should we also update the neighbourhood calculations to use e.g.
$$\min_{k \in K} [ c_k^{\text{dist}} d_{ij} + c_k^{\text{duration}} t_{ij}]$$
in the place of just $d_{ij}$? That'll definitely become more important with #475, but also already matters for HFVRP as well.

**Comment by N-Wouda** [comment]
Closed by #550.

**Comment by N-Wouda** [comment]
OK so we have algorithmic support for this already. But it's annoying having to work out all the fractions to round everything to integers for mixed objectives. So we might be better of switching to float/double first in #623, and then finalise this after.

**Comment by leonlan** [comment]
I prepared the HFVRP instances & best-known solutions in Sandbox. Once we have #623 then:
- [ ] Modify `read.py` to support the new HFVRP fields
- [ ] Move the instances over to PyVRP/Instances
- [ ] Setup benchmarks as well

**Comment by N-Wouda** [comment]
Since this is blocked by #623 I'm removing it from v0.10.0 as well.
